src/forms/datalist/DataList.tsx

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import {UID} from "react-uid";
+import {classNames} from "../../utils";
+import {Icon} from "../../icon";
+import {ClassNames} from "./ClassNames";
+
+/**
+ * General component description :
+ * DataList :
+ * The datalist element offers a flexible input when users need to filter
+ * and select from a large list of pre-defined options.
+ * Or, they need to input a custom value
+ * (not provided in the pre-defined list) for the input.
+ */
+
+/**
+ * Props for DataList option:
+ * @param {value} option value
+ * @param {selected} true if option is selected
+ * @param {style} CSS styles
+ * @param {className} CSS classname
+ **/
+type DataListOptionProps = {
+    value?: string;
+    selected?: boolean;
+    clasName?: string;
+    style?: any;
+};
+
+// Datalist option component
+export const DataListOption: React.FunctionComponent<DataListOptionProps> = ({
+    value,
+    selected,
+    children,
+    clasName,
+    style,
+}) => {
+    return (
+        <option value={value} selected={selected} className={ClassNames.NG_STAR_INSERTED} style={style}>
```

Comment:
Whats the use of NG_STAR_INSERTED here ?

Reply:
its a CSS class name :)